### ISSUE TITLE
Inverse affine transformation

### ISSUE THREAD

**Issue by old-school-kid** [keras-team/keras#16117]
If you open a GitHub issue, here is our policy:

It must be a bug, a feature request, or a significant problem with the documentation (for small docs fixes please send a PR instead).
The form below must be filled out.

**Here's why we have that policy:**.

Keras developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

**System information**.

TensorFlow version (you are using): 2.8
Are you willing to contribute it (Yes/No) : Yes

**Describe the feature and the current behavior/state**. 
Currently affine transformaton like `RandomFlip`, `RandomHeight`, `RandomRotation`, `RandomWidth`, `RandomCrop` and `RandomZoom` can only be used for image classification task since the ground truth for other visual tasks have to be changed for such augmentations.

Describe the feature clearly here. Be sure to convey here why the requested feature is needed. Any brief description about the use-case would help.
Want a layer(s) to reverse the affine transormation(s) so that the input can be augmentated for other visual taks.
**Will this change the current api? How?**
Yes. Addition of new layer(s).

**Who will benefit from this feature?**
Anyone using keras for visual tasks like segmentation, object detection etc

cc: @LukeWood

**Comment by qlzh727** [comment]
Thanks for reporting the issue.

We added a new base layer for https://github.com/keras-team/keras/blob/a179ed22f002e2f4a43ae4770348a9b8e1d5a051/keras/layers/preprocessing/image_preprocessing.py#L225. This will allow user to override `augment_label` and `augment_bounding_box`.

We are gradually applying the base layer to all the existing KPLs, and will make this base layer available publicly.

**Comment by mattdangerw** [comment]
Assigning to @qlzh727 as he is already holding the work to make this use case possible.

**Comment by tilakrayal** [comment]
@old-school-kid,
Could you please take a look at these two PR's which were already merged, `1.` Add support for **segmentation** mask in `BaseImageAugmentationLayer` `2. `Add all of the no-op segmentation mask augmentations.
https://github.com/keras-team/keras-cv/pull/748
https://github.com/keras-team/keras-cv/pull/751

Thank you!
